crates/biome_js_semantic/src/events.rs

```diff
@@ -556,33 +619,33 @@ impl SemanticEventExtractor {
 
         let infers = mem::take(&mut self.infers);
         for infer in infers {
-            let name_token = infer.ident_token().ok();
-            let parent_kind = infer.syntax().parent().map(|parent| parent.kind());
-
-            if let (Some(name_token), Some(parent_kind)) = (name_token, parent_kind) {
-                self.push_binding_into_scope(None, &name_token, parent_kind);
+            if let Ok(name_token) = infer.ident_token() {
+                let name = name_token.token_text_trimmed();
+                let name_range = name_token.text_range();
```

Comment:
Shouldn't we use `text_trimmed_range`?

Reply:
It is one of the changes I plan to propose in another PR because this propagates the change to the semantic model.